swiftnav/src/coords/ecef.rs

```diff
@@ -0,0 +1,255 @@
+use nalgebra::{ArrayStorage, Vector3};
+use std::ops::{Add, AddAssign, Mul, MulAssign, Sub, SubAssign};
+
+use crate::coords::LLHDegrees;
+
+use super::{AzimuthElevation, Ellipsoid, LLHRadians, NED, WGS84};
+
+/// WGS84 Earth Centered, Earth Fixed (ECEF) Cartesian coordinates (X, Y, Z).
+///
+/// Internally stored as an array of 3 [`f64`] values: x, y, z all in meters
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd, Default)]
+pub struct ECEF(Vector3<f64>);
+
+impl ECEF {
+    /// Create an [`ECEF`] object from the given X, Y, Z components
+    #[must_use]
+    pub fn new(x: f64, y: f64, z: f64) -> ECEF {
+        ECEF(Vector3::new(x, y, z))
+    }
+
+    /// Create an [`ECEF`] object from an array
+    ///
+    /// Element 0 is X, element 1 is Y, and element 2 is Z
+    #[must_use]
+    pub const fn from_array(array: &[f64; 3]) -> ECEF {
+        ECEF(Vector3::from_array_storage(ArrayStorage([*array; 1])))
+    }
+
+    /// Create an [`ECEF`] object from a [`nalgebra::Vector3<f64>`]
+    #[must_use]
+    pub(crate) fn from_vector3(vector: Vector3<f64>) -> ECEF {
+        ECEF(vector)
+    }
+
+    /// Get a reference to the inner array storing the data
+    #[must_use]
+    pub(crate) fn as_array_ref(&self) -> &[f64; 3] {
```

Comment:
Can we just call this `as_array`?

https://stackoverflow.com/questions/72996050/when-to-use-as-vs-to-vs-into-in-rust